File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    y1 = yellow # 노란색 영역 가로 길이
    y2 = 1 # 노란색 영역 세로 길이
    while True:
        if (y1+y2)*2 + 4 == brown:
            break
        else:
            temp = y2 + 1
            while temp * (yellow//temp) != yellow:
                temp += 1
            y2 = temp
            y1 = yellow//y2
        
    return [y1+2, y2+2]

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<pair<int, int>> pairs;
    for (int i = 1; i * i <= yellow; i++) {
        if (yellow % i == 0) {
            pairs.push_back({i, yellow / i});
        }
    }
    for (int i = 0; i < pairs.size(); i++) {
        int yellowH = pairs[i].first;
        int yellowW = pairs[i].second;
        if ((yellowH + 2) * (yellowW + 2) == brown + yellow) {
            return {yellowW + 2, yellowH + 2};
        }
    }
    return {0,0};
}
